atd-vzd/triggers/update_fatality_soft_delete.sql

```diff
@@ -0,0 +1,50 @@
+-- This function soft-deletes a fatality if the injury severity is changed from a 4 (killed).
+-- It undoes the soft-delete if it is changed back to 4. If it is a person record that
+-- doesn't yet exist in the fatalities table and the injury severity is changed to 4 then
+-- it will insert a new record into the fatalities table.
+CREATE OR REPLACE FUNCTION update_fatality_soft_delete()
+    RETURNS TRIGGER
+    LANGUAGE plpgsql
+AS $function$
+    BEGIN
+        -- if updated table was primary person
+    	IF (TG_TABLE_NAME = 'atd_txdot_primaryperson') THEN
+            -- if injury is updated to fatal
+        	IF (NEW.prsn_injry_sev_id = 4) THEN
+                -- if the fatality already exists then undo the soft-delete
+        	    IF (EXISTS (SELECT FROM fatalities WHERE primaryperson_id = NEW.primaryperson_id)) THEN
+                    UPDATE fatalities f
+                    SET is_deleted = false
+                    WHERE (f.primaryperson_id = NEW.primaryperson_id);
+                -- if the fatality doesnt exist yet add it to the table
+                ELSE
+                    INSERT INTO fatalities (crash_id, primaryperson_id)
+    		        VALUES (NEW.crash_id, NEW.primaryperson_id);
+    		    END IF;
```

Comment:
this works well :) 

if you wanted to get fancy, you could handle both the `if (exists)...else` block with an [upsert](https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-upsert/) into the fatalities table. in that case you'd insert but set `is_deleted = false` on conflict.

Reply:
TIL upsert, that looks a lot cleaner than a whole bunch of nested if/elses. implemented!